File: app/src/main/java/com/borisruzanov/russianwives/mvp/presenter/ChatMessagePresenter.java
package com.borisruzanov.russianwives.mvp.presenter;

import android.net.Uri;

import com.arellomobile.mvp.MvpPresenter;
import com.borisruzanov.russianwives.mvp.model.interactor.ChatMessageInteractor;
import com.borisruzanov.russianwives.mvp.view.ChatMessageView;

public class ChatMessagePresenter extends MvpPresenter<ChatMessageView> {

    private ChatMessageInteractor interactor;

    public ChatMessagePresenter(ChatMessageInteractor interactor) {
        this.interactor = interactor;
    }

    public void initChat(String friendUid){
        interactor.initChat(friendUid);
    }

    public void sendMessage(String friendUid, String message){
        interactor.sendMessage(friendUid, message, () -> getViewState().setEmptyMessage());
    }

    public void sendImage(String friendUid, Uri imageUri){
        interactor.sendImage(friendUid, imageUri, () -> getViewState().setEmptyMessage());
    }

}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/ui/chatmessage/ChatMessagePresenter.kt
package com.borisruzanov.russianwives.mvp.ui.chatmessage

import android.net.Uri

import com.arellomobile.mvp.InjectViewState
import com.arellomobile.mvp.MvpPresenter
import com.borisruzanov.russianwives.mvp.model.interactor.chatmessage.ChatMessageInteractor
import com.borisruzanov.russianwives.utils.UpdateCallback
import javax.inject.Inject

@InjectViewState
class ChatMessagePresenter @Inject constructor(private val interactor: ChatMessageInteractor)
    : MvpPresenter<ChatMessageView>() {

    fun sendMessage(friendUid: String, message: String) {
        interactor.sendMessage(friendUid, message, UpdateCallback{viewState.setEmptyMessage() })
    }

    fun sendImage(friendUid: String, imageUri: Uri) {
        interactor.sendImage(friendUid, imageUri, UpdateCallback { viewState.setEmptyMessage() })
    }

    fun removeOldMessage(chatUser: String?, id: String, message: String) {
        interactor.removeOldMessage(chatUser,id,message)
    }

}
